synapse/storage/databases/main/events.py

```diff
@@ -2071,13 +2069,18 @@ def _store_event_state_mappings_txn(
 
             state_groups[event.event_id] = context.state_group
 
-        self.db_pool.simple_insert_many_txn(
+        self.db_pool.simple_upsert_many_txn(
```

Comment:
Changed this to an upsert because to avoid colliding with a previous entry we put in via `store_state_group_id_for_event_id` earlier before we persisted the events.